src/core/src/op/reverse_sequence.cpp

```diff
@@ -59,25 +55,14 @@ shared_ptr<Node> op::ReverseSequence::clone_with_new_inputs(const OutputVector&
 
 void op::ReverseSequence::set_batch_axis(int64_t batch_axis) {
     m_batch_axis = batch_axis;
-    const auto& data_pshape = get_input_partial_shape(0);
-    const auto& data_rank = data_pshape.rank();
-    m_normalized_batch_axis = ov::normalize_axis(this, m_batch_axis, data_rank);
 }
 
-void op::ReverseSequence::set_sequence_axis(int64_t sequence_axis) {
-    m_seq_axis = sequence_axis;
-    const auto& data_pshape = get_input_partial_shape(0);
-    const auto& data_rank = data_pshape.rank();
-    m_normalized_seq_axis = ov::normalize_axis(this, m_seq_axis, data_rank);
+size_t op::ReverseSequence::get_batch_axis() const {
+    const auto& data_rank = get_input_partial_shape(0).rank();
+    return static_cast<size_t>(ov::normalize_axis(this, m_batch_axis, data_rank));
 }
 
-void op::ReverseSequence::normalize_axes(ReverseSequence* op) {
-    const auto& data_pshape = op->get_input_partial_shape(0);
-    const auto& data_rank = data_pshape.rank();
-    NODE_VALIDATION_CHECK(op,
-                          data_rank.is_dynamic() || data_rank.get_length() >= 2,
-                          "Data input rank should be equal or greater than 2. Got: ",
-                          data_pshape);
-    op->m_normalized_batch_axis = ov::normalize_axis(op, op->m_batch_axis, data_rank);
-    op->m_normalized_seq_axis = ov::normalize_axis(op, op->m_seq_axis, data_rank);
+void op::ReverseSequence::set_sequence_axis(int64_t sequence_axis) {
+    m_seq_axis = sequence_axis;
+    m_normalized_seq_axis = ov::normalize_axis(this, m_seq_axis, get_input_partial_shape(0).rank());
```

Comment:
`get_input_partial_shape(0)` makes impossible to call `set_sequence_axis` on default constructed op without inputs, 
can it be changed to perform normalization `if inputs.size() > 0`?
Also such normalization will throw for dynamic rank, but it wasn't supported anyway.

Reply:
using if there is no inputs is pointless and this value has no meaning for shape inference.